rustls/src/server/tls13.rs

```diff
@@ -570,12 +571,13 @@ mod client_hello {
     fn emit_hello_retry_request(
         transcript: &mut HandshakeHash,
         suite: &'static Tls13CipherSuite,
+        session_id: &SessionId,
```

Comment:
`SessionId` is `Copy`, just pass it by value?

Reply:
A good point for learning -- in C++ I would pass by const reference because a reference will fit in a register (or, failing that, not take a large amount of stack) and especially avoid passing by value where the value was bigger than, say, 64-bits. I guess rust's default ABI would avoid chucking the whole of the 32 bytes on the stack?

Reply:
I'm guessing it would actually go on the stack. Tried to do a Godbolt to verify but somehow doesn't work?

https://godbolt.org/z/fejKbrP65

Why would we want to avoid passing this array on the stack?

Reply:
Some ABIs can pass a small number of register-sized arguments in registers without touching the stack. In ARM APCS the magic number is 4 arguments.